esp_utils.py

```
@@ -65,28 +131,71 @@ def check_esp_idf_installed(idf_path: str = None) -> bool:
-        # Try to use idf.py to list ports (if available)
-        process = await asyncio.create_subprocess_shell(
-            "python -m serial.tools.list_ports",
-            stdout=asyncio.subprocess.PIPE,
-            stderr=asyncio.subprocess.PIPE
-        )
-        stdout, stderr = await process.communicate()
-        return process.returncode, stdout.decode(), stderr.decode()
+        # Try to list COM ports on Windows using mode command
+        if os.name == 'nt':  # Windows
+            result = subprocess.run(
+                ["mode"],
```

Ruff (0.14.14)

[error] Starting a process with a partial executable path

(S607)